src/main/java/org/generationcp/middleware/dao/dms/DmsProjectDao.java

```diff
@@ -1372,4 +1372,19 @@ public DatasetDTO getDatasetOfSampleList(final Integer sampleListId) {
 		}
 	}
 
+	public Integer getDatasetIdByStudyDbIdAndDatasetType(final Integer studyDbId, final DatasetTypeEnum datasetType) {
+		try {
+			final Query query = this.getSession().createSQLQuery("SELECT DISTINCT p.project_id"
+				+ " FROM project p "
+				+ " INNER JOIN nd_experiment nde ON nde.project_id = p.project_id"
+				+ " INNER JOIN project study ON p.study_id = study.project_id AND study.deleted != " + DELETED_STUDY
+				+ " WHERE nde.nd_geolocation_id = :studyDbId"
+				+ " AND p.dataset_type_id = " + datasetType.PLOT_DATA.getId());
```

Comment:
Use `query.setParameter` to set `datasetTypeId` (dataset type id). Also similarly, use `environmentId `as parameter name instead of `studyDbId`